db/buffered_transaction.go

```diff
@@ -10,13 +12,25 @@ type BufferedTransaction struct {
 	txn     Transaction
 }
 
+type BufferedTransactionWithAddress struct {
+	Txn     *BufferedTransaction
+	Address *felt.Felt
+}
+
```

Comment:
I don't think we should have this struct because it doesn't offer much functionality over BufferedTransaction, there is no new method being added.

Reply:
Will do this just you mentioned in above comments 👍